app/redux/store.js

```diff
@@ -11,17 +11,17 @@ const subscriber = ZMQ.socket('sub');
 requester.monitor(500, 0);
 subscriber.monitor(500, 0);
 
-const zmqMiddlewareCreator = (zmq, subzmq) => {
+const zmqMiddlewareCreator = (reqzmq, subzmq) => {
   return ({dispatch}) => {
     return next => action => {
-      if (!action.sock) {
+      if (!action.zmq) {
         return next(action);
       }
 
       console.log('zmqMiddleWare :: ', action);
 
-      const {type, sock, ...rest} = action;
-      console.log('zmqMiddleWare :: ', sock, type, rest);
+      const {type, zmq, ...rest} = action;
+      console.log('zmqMiddleWare :: ', zmq, type, rest);
 
       if (type) {
         next({type, ...rest});
```

Comment:
As discussed with @revathskumar this block can go towards the end. Its job is to pass on the data to the next middleware if any. Absence of this block breaks the middleware, however returning `next({type, ...rest})` as above in the case of `return next(action)` doesn't work. 

Also, we are yet to figure out the flow here, i.e, is control passed to the next middleware immediately? If so how is the execution of the remaining code in the current middleware and next middleware handled etc.

But to help with the flow, this block might be better placed towards the end.